distribution/client/src/lib/update.ts

```diff
@@ -126,20 +151,30 @@ export default class Update {
     }
 
     if (tasks.length == 0) {
-      logger.debug('No actionable tasks');
+      logger.warn('No actionable tasks');
       this.updateInProgress = null;
       this.saveUpdateSync(updates);
-      return false;
+      return true;
     }
 
+    logger.info(`Triggering update with tasks: ${tasks}`);
+
     return Promise.all(tasks).then(() => {
+
       UI.publish('All downloads completed, snapshotting data before restart');
       this.snapshotter.snapshot(`UL${this.getCurrentLevel()}->UL${updates.meta.level} Snapshot after downloads completed, before Jenkins restart`);
       this.saveUpdateSync(updates);
       UI.publish('All downloads completed and snapshotting done, restarting Jenkins');
-      this.restartJenkins();
+
+    }).then( () => {
+
```

Comment:
Mind removing some of these redundant newlines?

Reply:
Heh, I did that on purpose as it took me so long to reflect enough to find the right sequence of calls, that I wanted to make that sequence very readable.

But yeah I guess now it can be made slightly less verbose.